javascript/javascript1/week2/Candy.js

```diff
@@ -0,0 +1,73 @@
+// Candy helper optional 
+
+
+const boughtCandyPrice = []; 
+function addCandy( candyType , weight){
+   
+   let firstType ="Sweet";
+   let  secondType ="Chocolate";
+   let thirdType ="Toffee";
+   let fourthType = "Chewing-gum";
+    let price = 0; 
+    //const amountToSpend = Math.random() * 100; 
+
+    if ( firstType === "Sweet"){
+         price  = weight * 0.5;
+         console.log(boughtCandyPrice.push(price)); 
+    }
+    else if ( secondType === "Chocolate"){
+           price = weight * 0.7;
+
+        console.log( boughtCandyPrice.push(price));
+    }
+
+    else if ( thirdType === "Toffee"){
+          price= weight * 1.1;
+
+        console.log( boughtCandyPrice.push(price));
+    }
+    else if ( fourthType === "Chewing-gum"){
+      price= weight * 0.03;
+
+        console.log(boughtCandyPrice.push(price));
+    }
+    else {
+        console.log("wrong enter"); 
+    }
+
+}
+
+addCandy("Sweet", 30);
+addCandy("Toffee" ,40);
+addCandy("candy", 200); 
+console.log(boughtCandyPrice);
+
+
+
+// Can i buy more? 
+
+  const amountToSpend = Math.random() * 100; 
+  function canBuyMoreCandy(){
+    let total=0; 
+    let i=0; 
+    while( i < boughtCandyPrice.length){
+      i++; 
```

Comment:
there are 2 i++ in the body of the loop. Let's think how it works:
1. before the loop
i = 0
2. then we do i++ ( i is equal to 1) 
3. then we pick price from array by index 1 (but arrays are 0-indexed, so we're skipping the really first element in array) 
4. then we do i++ again (i is equal to 2) 
5. return back to the beginning of the loop 
6. i++ (i is equal to 3) 
7. pick price from array for index 3
8. and so on

With this approach we are taking price only for candies with odd indexes (1, 3, 5, 7, ...) and candies with even indexes (0, 2, 4, ...) are not taken into account